src/Ocelot/Configuration/CacheOptions.cs

```diff
@@ -1,15 +1,39 @@
-﻿namespace Ocelot.Configuration
+﻿using Ocelot.Request.Middleware;
+
+namespace Ocelot.Configuration
 {
     public class CacheOptions
-    {
-        public CacheOptions(int ttlSeconds, string region)
+    {
+        internal CacheOptions() { }
+
+        public CacheOptions(int ttlSeconds, string region, string header)
         {
             TtlSeconds = ttlSeconds;
-            Region = region;
+            Region = region;
+            Header = header;
+        }
+
+        public CacheOptions(int ttlSeconds, string region, string header, bool enableContentHashing)
+        {
+            TtlSeconds = ttlSeconds;
+            Region = region;
+            Header = header;
+            EnableContentHashing = enableContentHashing;
         }
 
-        public int TtlSeconds { get; }
-
-        public string Region { get; }
+        public int TtlSeconds { get; }
+        public string Region { get; }
+        public string Header { get; }
+
+        /// <summary>
+        /// Enables MD5 hash calculation of the <see cref="HttpRequestMessage.Content"/> of the <see cref="DownstreamRequest.Request"/> object.
+        /// </summary>
+        /// <remarks>
+        /// Default value is <see langword="false"/>. No hashing by default.
+        /// </remarks>
+        /// <value>
+        /// <see langword="true"/> if hashing is enabled, otherwise it is <see langword="false"/>.
+        /// </value>
+        public bool EnableContentHashing { get; }

```

Comment:
I think this was breaking change that causes wrong caches with POST requests with payload. Rolled back to version 21 where there is no issue. Haven't had time to confirm, but I'm 90% sure. Gonna try to add a unit test in a PR.